src/components/results/securityanalysis/security-analysis-result-tab.tsx

```diff
@@ -102,8 +102,25 @@ export const SecurityAnalysisResultTab: FunctionComponent<
         sortWay: SORT_WAYS.asc,
     });
 
+    const resultType = useMemo(() => {
+        if (tabIndex === N_RESULTS_TAB_INDEX) {
+            return RESULT_TYPE.N;
+        }
+
+        if (nmkType === NMK_TYPE.CONSTRAINTS_FROM_CONTINGENCIES) {
+            return RESULT_TYPE.NMK_CONTINGENCIES;
+        } else {
+            return RESULT_TYPE.NMK_LIMIT_VIOLATIONS;
+        }
+    }, [tabIndex, nmkType]);
+
+    const columnToFieldMapping = useMemo(
```

Comment:
Same, IMO you should directly call getColumnToFieldMapping, no need memo for this

Reply:
fixed

Reply:
No, I mean you should directly call getColumnToFieldMapping(resultType), otherwise with this fetchSecurityAnalysisResultWithQueryParams will be triggered at every rerender

Reply:
it's true, I fixed it too fast
I was memoizing it because it was used twice in the file, it was to avoid calling the method several times with the same exact params

But alright I'm changing it